discovery-provider/src/tasks/users.py

```diff
@@ -85,18 +87,43 @@ def user_state_update(
                     # Add or update the value of the user record for this block in user_events_lookup,
                     # ensuring that multiple events for a single user result in only 1 row insert operation
                     # (even if multiple operations are present)
-                    user_record = parse_user_event(
-                        self,
-                        user_contract,
-                        update_task,
-                        session,
-                        tx_receipt,
-                        block_number,
-                        entry,
-                        event_type,
-                        user_events_lookup[user_id]["user"],
-                        block_timestamp,
-                    )
+
+                    if event_type == user_event_types_lookup["update_multihash"]:
+                        metadata_multihash = helpers.multihash_digest_to_cid(
+                            entry["args"]._multihashDigest
+                        )
+                        user_record = (
+                            parse_user_event(
+                                self,
+                                user_contract,
+                                update_task,
+                                session,
+                                tx_receipt,
+                                block_number,
+                                entry,
+                                event_type,
+                                user_events_lookup[user_id]["user"],
```

Comment:
let's look up by txhash here AND user ids in order to facilitate multiple txs in a single block